accepted/00083-containerized-proxy.md

```diff
@@ -0,0 +1,87 @@
+    - one with cached content (`/var/cache/squid/`, `/var/cache/rhn`)
+    - one with log files (`/var/log/apache2`, `/var/log/squid`, `salt-broker` log)
+  - Proxy Containers can be started/stopped/exchanged - as long as the configuration directory stays the same the Proxy identity, history and functionality will be preserved. Optionally, the cached content directory can also be preserved for optimal performance
+  - Creating bootstrap scripts from containerized Proxies will not be possible. `mgr-bootstrap --hostname=$PROXY_NAME` from the Server will still be possible
+  - SSL certificates can be handled in three ways:
+    1. using the same CA as the Server, as generated at Server installation time. A new SSL certificate for the Proxy is generated at "configuration tarball" creation time. This will be the default option
+    2. using the same CA as the Server, as generated by the user ("bring your own SSL certs"). A new SSL certificate for the Proxy has to be placed in the configuration directory by the user
+    3. leave SSL termination to the user's infrastructure altogether
+
+## Limitations
+  - Only ZeroMQ Salt clients are supported
+  - No monitoring functionality
+  - No SLP functionality
+  - Traceback emails are not sent
+  - `rhn_package_manager` is not present nor supported
+
+## Later steps
+  - add Salt SSH support
+  - enable the use of "configuration directories" for VM Proxies as well
+  - Retail Branch Server: to be defined in a separate RFC
+
+## Other details
+  - Container image includes Apache, Proxy-specific Python WSGI applications and `salt-broker`
+  - base image will be Leap 15.3
+  - API/UI will need the following data to add a Proxy:
+    - FQDN
+    - http proxy hostname, username and password (optional, default from [ProxySettingsManager.getProxySettings()](https://github.com/uyuni-project/uyuni/blob/b7cf9270cc1c5edd9c15fa1217e280d645bfcef9/java/code/src/com/redhat/rhn/manager/setup/ProxySettingsManager.java#L42))
+    - choice for the SSL cert: generated or bring-your-own-certificate
+      1. if generated: [CNAMEs](https://en.wikipedia.org/wiki/CNAME_record), Common Name, Organization, OU, City, State, Country Code, e-mail, Password
+      2. if bring-your-own-certificate: nothing (with an explaination on where the files need to be placed should be displayed)
+      3. no SSL termination: nothing (with an explanation on having to provide termination via other means)
+  - Building of the image will happen in the Build Service, by and large leveraging existing packages
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+  - Long term this is a separate deployment mechanism to maintain and support. In future it should be decided whether keeping both is worth it
+
+# Alternatives
+[alternatives]: #alternatives
+
+ - Proxy could be split into more containers, according to the microservices pattern. This would make deployment and operations more complex, and no significant benefits have been highlighted so far
```

Comment:
In my eyes the benefit is that we don't have to have process management inside the container. Which makes them smaller and easier to move around. This is especially in edge scenarios indeed an advantage I believe. However I don't want to be the judge if this advantage is big enough. This would also mean that we should optimize the layers of our containers if we are thinking in such directions.

Reply:
Process management and smaller images: points taken, will expand them better.